speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl

```diff
@@ -140,6 +142,41 @@ tParticipantControl Mediator<DecisionMaker>::GetPControlSettings()
     return m_decisionMaker.Config.GetPControlSettings();
 }
 
+/// @brief  Sets the decision thresholds to what is in the xml file
+/// @param  p_filePath The path to the xml file
+/// @return The new set threshold values
+template <typename DecisionMaker>
+tThreshold Mediator<DecisionMaker>::SetThresholdSettings(const char* p_filePath)
+{
+    void* paramHandle = GfParmReadFile(p_filePath, GFPARM_RMODE_REREAD );
+
+    m_thresholds;
+    m_thresholds.Accel = GfParmGetNum(paramHandle, "Threshold Settings", "Accel", "%", STANDARD_THRESHOLD_ACCEL);
+    m_thresholds.Brake = GfParmGetNum(paramHandle, "Threshold Settings", "Brake", "%", STANDARD_THRESHOLD_BRAKE);
+    m_thresholds.Steer = GfParmGetNum(paramHandle, "Threshold Settings", "Steer", "%", STANDARD_THRESHOLD_STEER);
+
+    GfParmReleaseHandle(paramHandle);
+
+    m_thresholdsSet = true;
+    return m_thresholds;
```

Comment:
We want it to remember our settings, so we need to save it locally. Other settings can be sent to the SDAConfig when the menu is loaded, because you have to go through those menus to begin a race. This menu however is not always loaded, so the xml needs to be loaded elsewhere. I chose to load it in the mediator, that way decisions can load it through the mediator.

Reply:
The same thing happens with sounds (when starting the race, the xml is read), but those are used in the frontend, so they do not need to use the mediator